### PR TITLE
[INLONG-8951][Manager] Support for configuring built-in fields for iceberg and starrocks

### LINKED ISSUE #8951: [Improve][Manager] Support for configuring built-in fields for iceberg
### Description

Support for configuring built-in fields for iceberg.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
### Prepare a Pull Request

- Fixes #8951 

### Motivation

 Support for configuring built-in fields for iceberg.

### Modifications

 Support for configuring built-in fields for iceberg.

### CODE REVIEW

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/StarRocksLoadNode.java [hunk 1]
```diff
@@ -171,4 +175,20 @@ public String genTableName() {
...
+        if (!fieldNames.contains(MetaField.INLONG_DATA_TIME.name())) {
+            fieldInfos.add(0, new FieldInfo("inlong_data_time", new LongFormatInfo()));
+        }
+        return fieldInfos;
+    }
+
+    @Override
+    public List<FieldInfo> metaFields() {
+        List<FieldInfo> fieldInfos = new ArrayList<>();
+        fieldInfos.add(0, new FieldInfo("inlong_data_time", new LongFormatInfo()));
```
Comment (vernedeng) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/StarRocksLoadNode.java:191]: There is one constant named "inlong_data_time"

Comment (fuweng11) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/StarRocksLoadNode.java]: Done

**Comment by EMsnap** [comment]
image
please change this file also

**Comment by EMsnap** [comment]
image
should add this field also

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/StarRocksLoadNode.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fuweng11) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/StarRocksLoadNode.java]: Done.

**Review by vernedeng — APPROVED** [review]

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/IcebergProvider.java [hunk 2]
```diff
@@ -91,4 +98,22 @@ public LoadNode createLoadNode(StreamNode nodeInfo, Map<String, StreamField> con
                 icebergSink.getCatalogUri(),
                 icebergSink.getWarehouse());
     }
+
+    @Override
+    public List<StreamField> addStreamMetaFields(List<StreamField> streamFields) {
+        List<String> fieldNames = streamFields.stream().map(StreamField::getFieldName).collect(Collectors.toList());
+        if (!fieldNames.contains(MetaField.AUDIT_DATA_TIME.name())) {
+            streamFields.add(0,
+                    new StreamField(0, "long", MetaField.AUDIT_DATA_TIME.name(), "date_time", null, 1,
```
Comment (vernedeng) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/IcebergProvider.java:107]: data_time

Comment (fuweng11) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/node/provider/IcebergProvider.java]: Done.

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**